lang/java/reef-bridge-client/src/main/java/org/apache/reef/bridge/client/SecurityTokenHelper.java

```diff
@@ -0,0 +1,123 @@
+   * @return
+   * @throws IOException
+   */
+  static List<TokenInfo> getTokenInfo(final String securityTokenIdentifierFile,
+                                      final String securityTokenPasswordFile,
+                                      final String[] tokenKindString) throws IOException {
+
+    final List<TokenInfo> tokens = new ArrayList<>();
+    for (final String tokenStr : tokenKindString) {
+      LOG.log(Level.FINE, "tokenKind for each token:{0}.", tokenStr);
+      final String[] tokenParts = tokenStr.split(TOKEN_PART_SEPARATOR);
+      if (tokenParts.length != 2) {
+        LOG.log(Level.SEVERE, "Expected token parts length is 2 with format [tokenKind, tokenLen].");
+        throw new IllegalArgumentException("Expected token parts length is 2");
+      }
+
+      tokens.add(new TokenInfo(tokenParts[0], tokenParts[0], Integer.parseInt(tokenParts[1])));
+    }
+
+    try (final FileInputStream fStream = new FileInputStream(securityTokenPasswordFile)) {
+      try (final BufferedReader reader = new BufferedReader(new InputStreamReader(fStream, "UTF-8"))) {
+        for (final TokenInfo info : tokens) {
+          info.setPassword(reader.readLine());
+        }
+      }
+    }
+
+    try (final FileInputStream in = new FileInputStream(securityTokenIdentifierFile)) {
+      try (final DataInputStream din = new DataInputStream(in)) {
+        for (final TokenInfo info : tokens) {
+          final int len = info.getLength();
+          final byte[] b = new byte[len];
+          final int lenRead = din.read(b, 0, len);
+          LOG.log(Level.FINE, "Token length read: " + lenRead);
+          info.setKey(b);
+        }
+      }
+    }
+    return tokens;
+  }
+
+  /**
+   * Add a token to the currentUser.
+   *
+   * @param user - UserGroupInformation object.
+   * @param identifier - identifier of the token to be added.
+   * @param password - password of the token to be added.
+   * @param tokenKind - kind of the token to be added.
+   * @param tokenSerive - service of teh token to be added.
+   */
+  static void addToken(final UserGroupInformation user,
+                       final byte[] identifier,
+                       final byte[] password,
+                       final Text tokenKind,
+                       final Text tokenSerive) {
+    LOG.log(Level.FINE, "addToken for " + tokenKind);
+    final Token token = new Token(identifier, password, tokenKind, tokenSerive);
+    user.addToken(token);
+  }
+
+  /**
+   * Log all the tokens in the container for teh currentUser.
+   * Log info for debug purpose for now until it is stable then we can change the log level to FINE.
+   *
+   * @param user - the UserGroupInformation object.
+   * @param msgPrefix - msg prefix for log.
+   */
+  static void logToken(final UserGroupInformation user, final String msgPrefix) {
+    LOG.log(Level.INFO, msgPrefix + " number of tokens: [{0}].", user.getCredentials().numberOfTokens());
```

Comment:
`UserGroupInformation` already have a pretty decent `.toString()` implementation - can se use it in our logs?

Reply:
Also:
   * use string interpolation for logging;
   * check for `LOG.isLoggable()` before doing expensive logging;
   * use lower logging level (maybe pass it as a parameter, e.g. like we do in `ThreadLogger`)

Reply:
I tried UserGroupInformation .toString(), didn't get the information I need. The log here is for debug in E2E scenario for us to find what token that container would have. I will remove those lines after debug is done.